daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/CommandPreprocessor.scala

```diff
@@ -193,7 +193,7 @@ private[engine] class CommandPreprocessor(compiledPackages: ConcurrentCompiledPa
                   case Some(pkg) =>
                     PackageLookup.lookupRecord(pkg, recordId.qualifiedName) match {
                       case Left(err) => ResultError(err)
-                      case Right((dataTypParams, DataRecord(recordFlds, _mbTpl @ _))) =>
+                      case Right((dataTypParams, DataRecord(recordFlds, _))) =>
                         if (recordFlds.length != flds.length) {
```

Comment:
```suggestion
                        // note that we check the number of fields _before_ checking if we can do
                        // field reordering by looking at the labels. this means that it's forbidden to
                        // repeat keys even if we provide all the labels, which might be surprising
                        // since in JavaScript / Scala / most languages (but _not_ JSON, interestingly)
                        // it's ok to do `{"a": 1, "a": 2}`, where the second occurrence would just win.
                        if (recordFlds.length != flds.length) {
```